app/src/main/java/org/oppia/app/player/audio/LanguageDialogFragment.kt

```diff
@@ -6,69 +6,47 @@ import android.os.Bundle
 import androidx.appcompat.app.AlertDialog
 import androidx.fragment.app.DialogFragment
 import org.oppia.app.R
-import java.util.ArrayList
-
-private const val KEY_LANGUAGE_LIST = "LANGUAGE_LIST"
-private const val KEY_CURRENT_LANGUAGE = "CURRENT_LANGUAGE"
 
 /**
- * DialogFragment that controls language selection in audio and written translations
+ * DialogFragment that controls language selection in audio and written translations.
  */
 class LanguageDialogFragment : DialogFragment() {
   companion object {
-    lateinit var languageInterface1: LanguageInterface
+    lateinit var languageInterface: LanguageInterface
+    lateinit var languageList: List<String>
+    var selectedIndex: Int = 0
     /**
-     * This function is responsible for displaying content in DialogFragment
+     * This function is responsible for displaying content in DialogFragment.
      * @param languageInterface: [LanguageInterface] to send data back to parent
-     * @param languageCodeList: List of strings containing languages
+     * @param languageList: List of strings containing languages
      * @param currentLanguageCode: Currently selected language code
-     * @return LanguageDialogFragment: DialogFragment
+     * @return [LanguageDialogFragment]: DialogFragment
      */
     fun newInstance(
       languageInterface: LanguageInterface,
-      languageCodeList: List<String>,
+      languageList: List<String>,
       currentLanguageCode: String
     ): LanguageDialogFragment {
+      this.languageInterface = languageInterface
```

Comment:
Instead of sharing static state, why not save the value directly in the fragment? Though, I think you using Bundle arguments makes a bit more sense since then the fragment will retain these values upon configuration changes.

Reply:
I tried that Bundle arguments code but actually, it is not able to transfer that data to the `onCreateView` inside the dialogFragment and because of which I shifted to this. I will try that once again, and post the exact error.

Reply:
My point on static state was around using a companion object since that's equivalent to using static variables to pass state in Java. Using a Bundle mitigates this concern.

Aha, I think it's because you're using supportFragmentManager and not childSupportFragmentManager. We should probably use child fragment manager since that treats the current fragment as a parent rather than the fragment's parent.

Reply:
Thanks for pointing out this mistake. `childSupportFragmentManager ` works.